operator/api.go

```diff
@@ -37,16 +35,46 @@ func startServer(addr string, shutdownSignal <-chan struct{}, allSystemsGo <-cha
 		}
 	})
 
-	srv := &http.Server{Addr: addr}
+	errs := make(chan error, 1)
+	nServers := 0
 
-	go func() {
-		<-shutdownSignal
-		if err := srv.Shutdown(context.Background()); err != nil {
-			log.WithError(err).Error("apiserver shutdown")
+	// Since we are opening this on localhost only, we need to make sure
+	// we can open for both v4 and v6 localhost. In case the user is running
+	// v4-only or v6-only.
+	for _, addr := range addrs {
+		if addr == "" {
+			continue
 		}
-	}()
+		nServers++
+		srv := &http.Server{Addr: addr}
+		errCh := make(chan error, 1)
+
+		go func() {
+			err := srv.ListenAndServe()
+			if err != nil {
+				errCh <- err
+				errs <- err
+			}
+		}()
+		go func() {
+			select {
+			case <-shutdownSignal:
+				if err := srv.Shutdown(context.Background()); err != nil {
+					log.WithError(err).Error("apiserver shutdown")
+				}
+			case err := <-errCh:
+				log.Warnf("Unable to start status api: %s", err)
+			}
+		}()
+		log.Infof("Starting apiserver on address %s", addr)
+	}
 
-	log.Fatalf("Unable to start status api: %s", srv.ListenAndServe())
+	for err := range errs {
```

Comment:
Similar question as https://github.com/cilium/cilium/pull/10370#discussion_r385700594 (and I think whatever we do there will also work here).